run_nodes.sh

```diff
@@ -1,45 +1,82 @@
 #!/bin/bash
 
-if [ -z "$1" ] || (("$1" < 2 || "$1" > 8))
-then
-    echo "The committee size is missing, usage:
+function usage(){
+  echo "Usage:
+      ./run_nodes.sh [-v N_VALIDATORS] [-n N_NON_VALIDATORS] [-s] [ALEPH_NODE_ARG]...
+  where 2 <= N_VALIDATORS <= N_VALIDATORS + N_NON_VALIDATORS <= 10
+  (by default, N_VALIDATORS=4 and N_NON_VALIDATORS=0)"
+}
 
-    ./run_nodes.sh SIZE [Additional Arguments to ./target/debug/aleph-node]
+N_VALIDATORS=4
+N_NON_VALIDATORS=0
+BUILD_ALEPH_NODE='true'
 
-where 2 <= SIZE <= 8"
-    exit
-fi
+while getopts "v:n:s" flag
+do
+  case "${flag}" in
+    v) N_VALIDATORS=${OPTARG};;
+    n) N_NON_VALIDATORS=${OPTARG};;
+    s) BUILD_ALEPH_NODE='false';;
+    *)
+      usage
+      exit
+      ;;
+  esac
+done
+
+shift $((OPTIND-1))
 
 killall -9 aleph-node
 
 set -e
 
 clear
 
-n_members="$1"
-echo "$n_members" > /tmp/n_members
-shift
+if $BUILD_ALEPH_NODE ; then
+  cargo build --release -p aleph-node
+fi
+
+account_ids=(
+    "5D34dL5prEUaGNQtPPZ3yN5Y6BnkfXunKXXz6fo7ZJbLwRRH"
+    "5GBNeWRhZc2jXu7D55rBimKYDk8PGk8itRYFTPfC8RJLKG5o" \
+    "5Dfis6XL8J2P6JHUnUtArnFWndn62SydeP8ee8sG2ky9nfm9" \
+    "5F4H97f7nQovyrbiq4ZetaaviNwThSVcFobcA5aGab6167dK" \
+    "5DiDShBWa1fQx6gLzpf3SFBhMinCoyvHM1BWjPNsmXS8hkrW" \
+    "5EFb84yH9tpcFuiKUcsmdoF7xeeY3ajG1ZLQimxQoFt9HMKR" \
+    "5DZLHESsfGrJ5YzT3HuRPXsSNb589xQ4Unubh1mYLodzKdVY" \
+    "5GHJzqvG6tXnngCpG7B12qjUvbo5e4e9z8Xjidk3CQZHxTPZ" \
+    "5CUnSsgAyLND3bxxnfNhgWXSe9Wn676JzLpGLgyJv858qhoX" \
+    "5CVKn7HAZW1Ky4r7Vkgsr7VEW88C2sHgUNDiwHY9Ct2hjU8q")
+validator_ids=("${account_ids[@]::N_VALIDATORS}")
+# space separated ids
+validator_ids_string="${validator_ids[*]}"
+# comma separated ids
+validator_ids_string="${validator_ids_string//${IFS:0:1}/,}"
 
-# cargo build --release -p aleph-node
 
-authorities=(Damian Tomasz Zbyszko Hansu Adam Matt Antoni Michal)
-authorities=("${authorities[@]::$n_members}")
+echo "Bootstrapping chain for nodes 0..$((N_VALIDATORS - 1))"
+./target/release/aleph-node bootstrap-chain --base-path docker/data --chain-id dev --account-ids "$validator_ids_string" > docker/data/chainspec.json
```

Comment:
Suggestion about `chainspec.json` and `base-path` parameter: it would be nice to not modify folders that are under git governance by default. We already tackled some bug because one of our scripts was writing to that folder. Copy it somewhere else or something...

Reply:
done